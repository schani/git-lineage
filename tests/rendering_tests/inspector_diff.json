{
  "active_panel": "Inspector",
  "selected_file_navigator_index": null,
  "search_query": "",
  "in_search_mode": false,
  "commit_list": [
    {
      "hash": "xyz123abc456xyz123abc456xyz123abc456xyz1",
      "short_hash": "xyz123a",
      "author": "Dev User",
      "date": "10 mins ago",
      "subject": "Refactor code"
    }
  ],
  "selected_commit_index": 0,
  "current_content": [
    "def hello_world():",
    "    print('Hello, World!')",
    "",
    "def main():",
    "    hello_world()",
    "",
    "if __name__ == '__main__':",
    "    main()"
  ],
  "cursor_line": 4,
  "cursor_column": 8,
  "inspector_scroll_vertical": 0,
  "inspector_scroll_horizontal": 0,
  "show_diff_view": true,
  "status_message": "Diff view enabled",
  "is_loading": false,
  "file_tree_state": {
    "original_tree": {
      "root": [
        {
          "name": "example.py",
          "path": "./example.py",
          "is_dir": false,
          "git_status": "M",
          "is_expanded": false,
          "children": [],
          "parent_path": null
        }
      ],
      "repo_root": "",
      "git_status_map": {}
    },
    "display_tree": {
      "root": [
        {
          "name": "example.py",
          "path": "./example.py",
          "is_dir": false,
          "git_status": "M",
          "is_expanded": false,
          "children": [],
          "parent_path": null
        }
      ],
      "repo_root": "",
      "git_status_map": {}
    },
    "current_selection": "./example.py",
    "search_query": "",
    "in_search_mode": false
  },
  "active_file_context": "./example.py",
  "selected_commit_hash": "xyz123abc456xyz123abc456xyz123abc456xyz1"
}