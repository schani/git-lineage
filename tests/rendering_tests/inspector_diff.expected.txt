┌ File Navigator ──────────┐┌ Code Inspector (Diff View) ──────────────────────┐
│M example.py              ││   1 def hello_world():                           │
│                          ││   2     print('Hello, World!')                   │
│                          ││   3                                              │
│                          ││   4 def main():                                  │
│                          ││   5     hello_world()                            │
│                          ││   6                                              │
│                          ││   7 if __name__ == '__main__':                   │
│                          ││   8     main()                                   │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
└──────────────────────────┘│                                                  │
┌ Commit History (./example┐│                                                  │
│>> xyz123a 10 mins ago Dev││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
Diff view enabled | Tab: Switch panel | ↑↓: Navigate | p: Previous change | n: N
